packages/text-input/src/index.tsx

```diff
@@ -18,14 +18,19 @@ const TextInput: React.FC<TextInputProps> = ({
   ...props
 }) => {
   const defaults = useDefaults('textInput', props, {
+    width: undefined,
     height: 32,
     full: false,
     appearance: 'default' as TextInputAppearanceType
   })
   const inputRef = React.useRef<HTMLInputElement>(null)
 
   return (
-    <S.TextInputContainer ref={innerRef} {...defaults}>
+    <S.TextInputContainer
+      width={defaults.width}
+      full={defaults.full}
+      ref={innerRef}
```

Comment:
By the chance, can you make `className` prop to be attached to TextInputContainer instead of TextInput? otherwise it will be difficult to style container using TextInputElements due of reversed descending.